### PR TITLE
Better error handling for RSVPs + NB client refactor

### PR DESCRIPTION
**Pull request by kyleschmolze**
Ok, here's an attempted fix for the issue I addressed in #168 

1. Start off the rsvp action with a `match` call first, and if we find the existing user, just use that for the RSVP. Only if there isn't an existing user should we "push". This should prevent the mentioned bug from happening, and allows us to property capture validation errors from the response.
2. Re-org `rescue` calls to be slightly more aesthetically appealing to yours truly by removing unnecessary `begin` statements.
3. Setup `$nation_builder_client` global variable in an initializer so we don't have duplicated creation-code all over the repo.

### CODE REVIEW

File: config/initializers/nation_builder.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+if ENV['NATION_NAME'] && ENV['NATION_API_TOKEN']
+  $nation_builder_client = NationBuilder::Client.new(ENV['NATION_NAME'], ENV['NATION_API_TOKEN'])
```
[the remark this answers is not in the document]
Comment (tilgovi) [reply config/initializers/nation_builder.rb]: Is it worth doing anything in an `else` clause to provide greater feedback when starting without the proper environment variables set? Before, there were the checks in the other places the client was initialized. Now, it'll just be an undefined symbol for `$nation_builder_client`, yeah?

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class EventsController < ApplicationController
   before_action :require_nationbuilder_slug
+  before_action :set_event, only: [:show, :rsvp]
```
Comment (tilgovi) [inline app/controllers/events_controller.rb]: 👏

File: config/initializers/nation_builder.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (kyleschmolze) [reply config/initializers/nation_builder.rb]: This is pretty common, yes.

As for the error cases, it's not clear to me what should happen when no nation builder client is initialized. Either way we'll probably want to throw an error if you try to view the calendar, but right now the error will be an "undefined method for nil". If we wanted to be more helpful, I guess we could add a simple `before_action` which checks for the existence of the variable and throws a more helpful error.

[the remark this answers is not in the document]
Comment (tilgovi) [reply config/initializers/nation_builder.rb]: Probably fine as is, I just wanted to point it out.

**Comment by dominic** [comment]
I totally missed this between meetings. I'm gonna test this out now and maybe merge in advance of another NB change :+1:

File: config/initializers/nation_builder.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply config/initializers/nation_builder.rb]: The `$` prefix isn't just a convention for globals, it's a [language feature](https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/globalvars.html).

[the remark this answers is not in the document]
Comment (kyleschmolze) [reply config/initializers/nation_builder.rb]: ooh thanks for the scripture, @eventualbuddha, had no idea about tracing or those special globals.

**Pull request merged**